includes/amp-helper-functions.php

```diff
@@ -334,6 +334,16 @@ function amp_add_generator_metadata() {
  */
 function amp_register_default_scripts( $wp_scripts ) {
 
+	// Polyfill dependencies that are registered in Gutenberg and WordPress 5.0.
+	if ( ! has_action( 'wp_default_scripts', 'gutenberg_register_packages_scripts' ) ) {
```

Comment:
I think it's okay having this, but strictly it isn't needed right? It only covers the plugin anyway I think?

Reply:
How that? Shouldn't we just _always_ want to check whether these two scripts are registered, and if not do so ourselves?

Reply:
That being said, `gutenberg_register_packages_scripts()` explicitly accounts for this case in `gutenberg_register_packages_scripts()` by [calling `gutenberg_override_script()`](https://github.com/WordPress/gutenberg/blob/80be381921ce818c0381e877f510bbc0be0fcf24/lib/client-assets.php#L144-L159) which does `wp_deregister_script()` first. So it looks like the condition is indeed unnecessary.